Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +15,72 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
+        }
+
+        public static int FindDigit(string equation)
+        {
+            // Splitting the equation into three parts A,B,C
+
+            List<string> tokens = new List<string>();
+            tokens = equation.Split('=','*').ToList();
+            var A = tokens[0];
+            var B = tokens[1];
+            var C = tokens[2];
+
+            //null check on tokens
+            if(A==null || B==null || C==null)
+                return -1;
+
+            // First checking if C contains ? character 
+
+            if(C.Contains("?")){
+                var numeric_A = Convert.ToInt32(A);
+                var numeric_B = Convert.ToInt32(B);
+               
+                // FindDigit method will compare two strings  for different characters or length */
+                return FindDigit(Convert.ToString(numeric_A * numeric_B),C);
+            }
+
+            // Checking for A only if it contains '?' if B contains then swap B with A */
+
+            else{
+                if(B.Contains("?"))
+                    Swap(ref A,ref B);
+                    
+                var numeric_C = Convert.ToInt32(C);
+                var numeric_B = Convert.ToInt32(B);
+
+                if(numeric_C % numeric_B !=0 || numeric_B == 0)
+                    return -1;
+                
+                 // FindDigit method will compare two strings  for different characters or length 

```

Comment:
Move this comment to method definition.
Also as the name suggest, FindDigit should find missing digit based on certain conditions (which you should specify in your comment), comment you wrote is saying that FindDigit will compare 2 strings for different characters and length....???